components/kyma-environment-broker/internal/suspension/handler.go

```diff
@@ -63,30 +65,47 @@ func (h *ContextUpdateHandler) Handle(instance *internal.Instance, newCtx intern
 	}
 }
 
-func (h *ContextUpdateHandler) suspend(instance *internal.Instance, log logrus.FieldLogger) error {
+func (h *ContextUpdateHandler) suspend(instance *internal.Instance, log logrus.FieldLogger) (string, error) {
 	lastDeprovisioning, err := h.operations.GetDeprovisioningOperationByInstanceID(instance.InstanceID)
 	// there was an error - fail
 	if err != nil && !dberr.IsNotFound(err) {
-		return err
+		return "", err
 	}
 
-	// no error, operation exists and is in progress
+	// no error, deprovisioning operation exists and is in progress
 	if err == nil && (lastDeprovisioning.State == domain.InProgress || lastDeprovisioning.State == orchestration.Pending) {
-		log.Infof("Suspension already started")
-		return nil
+		if !lastDeprovisioning.Temporary {
+			// found ordinary deprovisioning operation in progress - fail suspension
+			log.Warnf("Deprovisioning operation already started for instance %s", instance.InstanceID)
+			return "", errors.New("cannot suspend instance - deprovisioning operation already started")
```

Comment:
I think we should not return an error here. When the instance is being deprovisioned we should not do anything. The context update response could be HTTP 200